java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/OAuthRefreshTokenSourceAuthStrategy.java

```diff
@@ -384,11 +387,18 @@ protected boolean shouldRefresh(AccessToken accessToken, Instant now) {
                 return true;
             }
             Instant expiresAt = accessToken.getExpirationTime().toInstant();
-            Instant thresholdToProactiveRefresh = expiresAt
-                .minusSeconds(randomNumberGenerator.nextInt((int) MAX_PROACTIVE_TOKEN_REFRESH.toSeconds()));
+            Instant thresholdToProactiveRefresh = expiresAt.minusSeconds(getProactiveGracePeriodSeconds());
             return now.isAfter(thresholdToProactiveRefresh);
         }
 
+        @VisibleForTesting
+        protected int getProactiveGracePeriodSeconds() {
+            int maxSeconds = (int) MAX_PROACTIVE_TOKEN_REFRESH.toSeconds();
+            int minSeconds = (int) MIN_PROACTIVE_TOKEN_REFRESH.toSeconds();
+            Preconditions.checkArgument(maxSeconds > minSeconds);
```

Comment:
well, not needed at runtime ... could do it in the tests. 

and could simple test `getProactiveGracePeriodSeconds` a bunch of times and ensure always > 0